share/availability/light/testing.go

```diff
@@ -0,0 +1,56 @@
+package light
+
+import (
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/celestia-node/share/service"
+
+	"github.com/ipfs/go-blockservice"
+	mdutils "github.com/ipfs/go-merkledag/test"
+	routinghelpers "github.com/libp2p/go-libp2p-routing-helpers"
+	"github.com/libp2p/go-libp2p/p2p/discovery/routing"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/availability/discovery"
+	availability_test "github.com/celestiaorg/celestia-node/share/availability/test"
+)
+
+// randLightServiceWithSquare provides a share.Service filled with 'n' NMT
+// trees of 'n' random shares, essentially storing a whole square.
+func randLightServiceWithSquare(t *testing.T, n int) (*service.ShareService, *share.Root) {
+	bServ := mdutils.Bserv()
+
+	return service.NewShareService(bServ, TestLightAvailability(bServ)), availability_test.RandFillBS(t, n, bServ)
+}
+
+// randLightService provides an unfilled share.Service with corresponding
+// blockservice.BlockService than can be filled by the test.
+func randLightService() (*service.ShareService, blockservice.BlockService) {
+	bServ := mdutils.Bserv()
+	return service.NewShareService(bServ, TestLightAvailability(bServ)), bServ
+}
+
+// randLightNode creates a Light Node filled with a random block of the given size.
+func randLightNode(dn *availability_test.DagNet, squareSize int) (*availability_test.Node, *share.Root) {
```

Comment:
Public please

Reply:
Yeah, might be reused somewhere else eventually. Made it public